runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -1337,32 +1236,29 @@ JITServerAOTCache::readCache(FILE *f, const JITServerAOTCacheHeader &header, TR_
    Vector<AOTCacheWellKnownClassesRecord *> wellKnownClassesRecords(header._nextWellKnownClassesId, NULL, stackMemoryRegion);
    Vector<AOTCacheAOTHeaderRecord *> aotHeaderRecords(header._nextAOTHeaderId, NULL, stackMemoryRegion);
 
-   if (!readRecords(f, header._numClassLoaderRecords, _classLoaderMap, _classLoaderHead, _classLoaderTail, classLoaderRecords,
-                    classLoaderRecords, classRecords, methodRecords, classChainRecords, wellKnownClassesRecords, aotHeaderRecords))
+   JITServerAOTCacheReadContext context = {f, classLoaderRecords, classRecords, methodRecords,
```

Comment:
I think it might be cleaner to store the vectors themselves instead of references in `JITServerAOTCacheReadContext` and initialize them in its constructor from the `TR::Region`.